pkg/gardener/shoot/extender/restrictions/restrictions.go

```diff
@@ -0,0 +1,42 @@
+package restrictions
+
+import (
+	gardener "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
+)
+
+const (
+	euAccessAddons = "support.gardener.cloud/eu-access-for-cluster-addons"
+	euAccessNodes  = "support.gardener.cloud/eu-access-for-cluster-nodes"
+)
+
+func ExtendWithAccessRestriction() func(runtime imv1.Runtime, shoot *gardener.Shoot) error {
+	return func(runtime imv1.Runtime, shoot *gardener.Shoot) error {
+		if isEuAccess(runtime.Spec.Shoot.PlatformRegion) {
+			shoot.Spec.AccessRestrictions = append(shoot.Spec.AccessRestrictions, extendWithEuAccess())
+		}
+		return nil
+	}
+}
+
+func extendWithEuAccess() gardener.AccessRestrictionWithOptions {
```

Comment:
We can avoid creating intermediate collection and simplify, by passing shoot pointer here and changing function to: 

```golang
func extendWithEuAccess(shoot *gardener.Shoot) {
	shoot.Spec.AccessRestrictions = append(shoot.Spec.AccessRestrictions, gardener.AccessRestrictionWithOptions{
		AccessRestriction: gardener.AccessRestriction{
			Name: "eu-access-only",
		},
		Options: map[string]string{
			euAccessAddons: "true",
			euAccessNodes:  "true",
		},
	})
}
```